R/FilteredDatasetDefault.R

```diff
@@ -1,269 +1,107 @@
+        trim_position <- 37L
+        class_string <- strtrim(class_string, trim_position)
+        substr(class_string, 35L, 37L) <- "..."
       }
-      logger::log_trace("FilteredDatasetDefault$get_call initializing for dataname: { private$dataname }")
-      filter_call
+      sprintf(" - unfiltered dataset:\t\"%s\":   %s", private$dataname, class_string)
     },
 
-    #' @description
-    #' Set filter state
-    #'
-    #' @param state (`teal_slice`) object
-    #'
-    #' @examples
-    #' dataset <- teal.slice:::DefaultFilteredDataset$new(iris, "iris")
-    #' fs <- teal_slices(
-    #'   teal_slice(dataname = "iris", varname = "Species", selected = "virginica"),
-    #'   teal_slice(dataname = "iris", varname = "Petal.Length", selected = c(2.0, 5))
-    #' )
-    #' dataset$set_filter_state(state = fs)
-    #' shiny::isolate(dataset$get_filter_state())
-    #'
-    #' @return `NULL` invisibly
-    #'
+    #' @param sid (`character`) for method consistency, ignored
+    #' @return `NULL`, invisibly.
+    get_call = function(sid) {
+      invisible(NULL)
+    },
+    #' @return `NULL`, invisibly.
+    get_filter_state = function() {
+      invisible(NULL)
+    },
+    #' @param state (`teal_slices`) for method consistency, ignored
+    #' @return `NULL`, invisibly.
     set_filter_state = function(state) {
-      shiny::isolate({
-        logger::log_trace("{ class(self)[1] }$set_filter_state initializing, dataname: { private$dataname }")
-        checkmate::assert_class(state, "teal_slices")
-        lapply(state, function(slice) {
-          checkmate::assert_true(slice$dataname == private$dataname)
-        })
-        private$get_filter_states()[[1L]]$set_filter_state(state = state)
-        invisible(NULL)
-      })
+      if (length(state) != 0L) {
+        warning("DefaultFilterState cannot set state")
+      }
+      invisible(NULL)
+    },
+    #' @param force (`logical`) for method consistency, ignored
+    #' @return `NULL`, invisibly.
+    clear_filter_states = function(force) {
+      invisible(NULL)
     },
 
-    #' @description
-    #' Remove one or more `FilterState` form a `FilteredDataset`
-    #'
-    #' @param state (`teal_slices`)\cr
-    #'   specifying `FilterState` objects to remove;
-    #'   `teal_slice`s may contain only `dataname` and `varname`, other elements are ignored
-    #'
-    #' @return `NULL` invisibly
-    #'
-    remove_filter_state = function(state) {
-      checkmate::assert_class(state, "teal_slices")
-
-      shiny::isolate({
-        logger::log_trace("{ class(self)[1] }$remove_filter_state removing filter(s), dataname: { private$dataname }")
-
-        varnames <- unique(unlist(lapply(state, "[[", "varname")))
-        private$get_filter_states()[[1]]$remove_filter_state(state)
-
-        logger::log_trace("{ class(self)[1] }$remove_filter_state removed filter(s), dataname: { private$dataname }")
-      })
-
+    #' @return `NULL`, invisibly
+    get_filter_overview = function() {
```

Comment:
WDYT? This could be improved in another issue. Like adding an icon like 'fa-triangle-exclamation' with [popover](https://getbootstrap.com/docs/4.0/components/popovers/) displaying message "not supported in the filter panel"

```suggestion
    get_filter_overview = function() {
      data.frame(dataname = private$dataname)
```
<img width="354" alt="Skärmavbild 2023-12-14 kl  14 06 20" src="https://github.com/insightsengineering/teal.slice/assets/6959016/da30ff9a-626e-475b-957d-0c6827a3729b">

Reply:
I did consider displaying unfiltered datasets here, but then again I explicitly removed them from `Active Filters`.
These objects are not being filtered, so I don't know if displaying them in _filter_ summary makes sense.

A more general question: Do we acknowledge them at all or hide them from view entirely?

Reply:
They are datasets which could be used by specific module. I think it is worth it to display what dataset is in this module.